### ISSUE TITLE
Ethereum failure

### ISSUE THREAD

**Issue by claudioandre-br** [magnumripper/JohnTheRipper#3344]
This is clang. I tried to reproduce on super, but it doesn't have clang.

```
form=ethereum                     guesses:    4 -show=   4 0:00:00:00 DONE : Expected count(s) (7)(-show7)  [!!!FAILED1!!!  ret_val=0]
.pot CHK:ethereum                 guesses:    4 0:00:00:00 DONE  [PASSED] (4 val-pwd)
```

- missing a buffer cleanup somewhere?
- I can see some failures on pbkdf2-hmac-* (on super).

**Comment by magnumripper** [comment]
Doesn't well have clang? Or was that bull?

Several pbkdf2-hmac- formats have "false positives" when using TS `-internal` but this is not that, right?

**Comment by claudioandre-br** [comment]
Not sure about both questions. 

Anyway, if you can review your patch trying to identify an uninitialized buffer (or upper > 64 buffer), or something that causes random failures.

**Comment by magnumripper** [comment]
My patch? What patch?

**Comment by claudioandre-br** [comment]
I can try to bisect, using CI. The random issues are very frequent.

**Comment by claudioandre-br** [comment]
I was able to create a reproducer: `while ./jtrts.pl -internal --type ethereum; do :; done`

On super (less than 1 hour)
```
Running JTRTS in -internal mode

form=ethereum                     guesses:    6 -show=   6 0:00:00:00 DONE : Expected count(s) (7)(-show7)  [!!!FAILED1!!!  ret_val=0]
../run/unique: /usr/local/cuda-8.0/targets/x86_64-linux/lib/libOpenCL.so.1: no version information available (required by ../run/unique)
.pot CHK:ethereum                 guesses:    6 0:00:00:00 DONE  [PASSED] (6 val-pwd)
Some tests had Errors. Performed 1 tests.  1 errors
Time used was 27 seconds
```

**Comment by magnumripper** [comment]
> I was thinking about a620d61c3f8fc2b46551f92526e1183701aebcc0.
>
> A new `include`, setting `PLAINTEXT_LENGTH`, ... Anyway, it was a guess. I can't say for sure.

That's fine but that commit didn't touch any file that Ethererum-opencl is using. That change was all CPU-side and isolated to two formats, no shared code was touched.

> I was able to create a reproducer: `while ./jtrts.pl -internal --type ethereum; do :; done`
>
> On super (less than 1 hour)

Ouch, so it's intermittent? We'll need lots of whisky.

**Comment by claudioandre-br** [comment]
It seems I'm found two different issues. One of them is related to a job timeout after recent commits:

To fix it I reverted the last 3 commits and I get back 5-10 minutes. So, if the new tests make testing 10 minutes slower, that is ok.
https://github.com/claudioandre-br/JohnTheRipper/compare/d730efef379e...225bbd475da2
****

The other is `while ./jtrts.pl -internal --type ethereum; do :; done`. It is hard to say, but I would say it was not here a week ago.

**Comment by magnumripper** [comment]
> Regarding the job timeout issue, it seems that benchmarking the DMG formats is super slow. I am not sure if recent commits changed something related to DMG benchmarking.

Sure, three very large test vectors were added. Perhaps we should regenerate them using DMG v1 and shorter data.

**Comment by magnumripper** [comment]
Also, we only really need one single test vector with length > 64 so we could drop two of them.

**Comment by claudioandre-br** [comment]
No, it is not new. I reverted to c555cfa2b3675b54e47b4cd1f3adfffb0843de09 and it still fails in 10min:

```
John Jumbo build detected.
Running JTRTS in -internal mode

form=ethereum                     guesses:    6 -show=   6 0:00:00:00 DONE : Expected count(s) (7)(-show7)  [!!!FAILED1!!!  ret_val=0]
.pot CHK:ethereum                 guesses:    6 0:00:00:00 DONE  [PASSED] (6 val-pwd)
Some tests had Errors. Performed 1 tests.  1 errors
```

**Comment by kholia** [comment]
I am on mobile phone now.

I recently added SIMD support to the Presale Wallet code in this format. Is that causing problems?

Can we narrow down the failure to a particular hash?

What happens when we enable ASan and run the test suite in a loop? (Disable memory leak checking using an ASan runtime option).

**Comment by magnumripper** [comment]
I mistakenly always thought it was ethereum-opencl failing, but this is the CPU format. I'll review latest changes now.

**Comment by magnumripper** [comment]
We should test commit before f65afc1f and then commit before b36b0676, then baa9dfd5

**Comment by magnumripper** [comment]
f65afc1 "Add SIMD support for cracking Ethereum Presale wallets" failed within a few minutes. The commit before it doesn't seem to fail. I can't see anything obviously wrong with that commit though.

**Comment by magnumripper** [comment]
OK I had `f65afc1^` fail now, so that one wasn't it.

**Comment by magnumripper** [comment]
Now testing commit right before baa9dfd

**Comment by claudioandre-br** [comment]
> I mistakenly always thought it was ethereum-opencl failing, but this is the CPU format. I'll review latest changes now.

super complaining about OpenCL doesnt help (I disabled it during the second test to make sure I was not doing wrong). But, you did a hurried reading. 
****

Well, I have some PBKDF2 formats disabled during testing for a looong time. Your findings do not surprise me.

**Comment by magnumripper** [comment]
`baa9dfd^` didn't fail in 10 hours. So the problem was introduced with b36b067 + baa9dfd (they should have been one single commit).

> I have some PBKDF2 formats disabled during testing for a looong time

Many PBKDF2 formats (as well as bcrypt and perhaps some other) fail the TS `-internal` test. This is merely a shortcoming of TS. They have some identical test vectors that look differently and TS can't tell.
